packages/block-editor/src/components/rich-text/content.js

```diff
@@ -15,33 +15,40 @@ import RichText from './';
  */
 import { getMultilineTag } from './utils';
 
-export function Content( {
-	value,
-	tagName: Tag,
-	multiline,
-	format,
-	...props
-} ) {
+export function valueToHTMLString( value, multiline ) {
 	if ( RichText.isEmpty( value ) ) {
-		const MultilineTag = getMultilineTag( multiline );
-		value = MultilineTag ? <MultilineTag /> : null;
-	} else if ( Array.isArray( value ) ) {
+		const multilineTag = getMultilineTag( multiline );
+		return multilineTag ? `<${ multilineTag }></${ multilineTag }>` : '';
+	}
+
+	if ( Array.isArray( value ) ) {
 		deprecated( 'wp.blockEditor.RichText value prop as children type', {
 			since: '6.1',
 			version: '6.3',
 			alternative: 'value prop as string',
 			link: 'https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/introducing-attributes-and-editable-fields/',
 		} );
-		value = <RawHTML>{ childrenSource.toHTML( value ) }</RawHTML>;
-	} else if ( typeof value === 'string' ) {
-		// To do: deprecate.
-		value = <RawHTML>{ value }</RawHTML>;
-	} else {
-		// To do: create a toReactComponent method on RichTextData, which we
-		// might in the future also use for the editable tree. See
-		// https://github.com/WordPress/gutenberg/pull/41655.
-		value = <RawHTML>{ value.toHTMLString() }</RawHTML>;
+		return childrenSource.toHTML( value );
+	}
+
+	// To do: deprecate string type.
+	if ( typeof value === 'string' ) {
+		return value;
 	}
 
+	// To do: create a toReactComponent method on RichTextData, which we
+	// might in the future also use for the editable tree. See
+	// https://github.com/WordPress/gutenberg/pull/41655.
+	return value.toHTMLString();
+}
+
+export function Content( {
+	value,
+	tagName: Tag,
+	multiline,
+	format,
```

Comment:
@ellatrix, is the `format` prop used anywhere?

Reply:
It's a very old prop. I believe years ago at the start of GB we had `string` and `children`. We don't check it anymore, instead we rely on the type of the value (string or array). But we also shouldn't remove this because we'd be spreading it on a plain tag. Maybe it's ok because it's just a React warning in dev mode?

Reply:
Got it; it seems fine to keep it around.

There are no warnings. I was looking into React Native code and noticed this prop, but I wasn't sure if it was used anywhere.